[stable/oauth2-proxy] Adds additional configuration options for oauth2 providers

tombotch opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

**What this PR does / why we need it**:
This PR adds configuration for the `google` and `github` providers. This allows authentication to easily be restricted to certain groups, orgs and teams. The `google` provider configuration specifically adds the required credentials to the existing oauth2_proxy secret.

It also moves the e-mail domains configuration to its own field within `values.yaml`, which allows these to more easily be configured within a values file.

**Special notes for your reviewer**:
I've added some additional checks in to ensure that backwards compatibility is maintained.

tombotch committed:
Adds additional provider configuration

This adds additional configuration options for the GitHub and Google providers, to allow authentication to be restricted to certain groups.

It also moves the e-mail domains configuration to a specific option in the values file.

tombotch committed:
Fixes typo and adds conditional check to emailDomains field

tombotch committed:
Allows e-mail domains to be overriden in extraArgs

This is for backwards compatibility

compleatang commented on stable/oauth2-proxy/templates/deployment.yaml:
```diff
@@ -34,6 +34,26 @@ spec:
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         args:
+        {{- if (not (index .Values.extraArgs "email-domain")) and (.Values.emailDomains) }}
```
for readability IMV it's best to outdent `if`, `range` and `end` statements.

compleatang commented on stable/oauth2-proxy/values.yaml:
```diff
@@ -13,6 +17,40 @@ config:
   #   pass_access_token = true
   configFile: ""
 
+# Provider-specific configuration
```
personally, i like this method. but it needs documenting in the README.

tombotch committed:
Updates README and version

This also changes the indentation of the if and range statements for readability

tombotch committed:
Fixes linting issue
